custom/icds_reports/utils/aggregation_helpers/distributed/agg_gov_dashboard.py

```diff
@@ -0,0 +1,194 @@
+            month = self.month_start
+        return "{}_{}".format(self.base_tablename, month.strftime("%Y-%m-%d"))
+
+    def drop_table_if_exists(self):
+        return """
+        DROP TABLE IF EXISTS "{table_name}"
+        """.format(table_name=self._tablename_func())
+
+    def create_child_table(self):
+        return """
+        CREATE TABLE "{table_name}" (
+        CHECK (month=DATE %(start_date)s)
+        ) INHERITS ({base_tablename})
+        """.format(
+            table_name=self._tablename_func(),
+            base_tablename=self.base_tablename,
+        ), {
+            "start_date": self.month_start
+        }
+
+    def aggregate_query(self):
+        """
+        Returns the base aggregate query which is used to insert all the locations
+        into the LS data table.
+        """
+
+        columns = (
+            ('state_id', 'awc_location_local.state_id'),
+            ('district_id', 'awc_location_local.district_id'),
+            ('block_id', 'awc_location_local.block_id'),
+            ('supervisor_id', 'awc_location_local.supervisor_id'),
+            ('awc_id', 'awc_location_local.doc_id'),
+            ('awc_site_code', 'awc_location_local.awc_site_code'),
+            ('month', "'{}'".format(month_formatter(self.month_start))),
+            ('awc_launched', 'agg_awc.num_launched_awcs=1'),
+            ('total_preg_benefit_till_date', 'COALESCE(agg_awc.cases_ccs_pregnant, 0)'),
+            ('total_lact_benefit_till_date', 'COALESCE(agg_awc.cases_ccs_lactating, 0)'),
+            ('total_preg_reg_till_date', 'COALESCE(agg_awc.cases_ccs_pregnant_all,0)'),
+            ('total_lact_reg_till_date', 'COALESCE(agg_awc.cases_ccs_lactating_all,0)'),
+            ('total_lact_benefit_in_month', 'COALESCE(agg_awc.cases_ccs_lactating_reg_in_month,0)'),
+            ('total_preg_benefit_in_month', 'COALESCE(agg_awc.cases_ccs_pregnant_reg_in_month,0)'),
+            ('total_lact_reg_in_month', 'COALESCE(agg_awc.cases_ccs_lactating_all_reg_in_month,0)'),
+            ('total_preg_reg_in_month', 'COALESCE(agg_awc.cases_ccs_pregnant_all_reg_in_month,0)')
+        )
+        return """
+        INSERT INTO "{tablename}" (
+            {columns}
+        )
+        (
+        SELECT
+        {calculations}
+        from awc_location_local  LEFT JOIN agg_awc  ON (
+            awc_location_local.doc_id = agg_awc.awc_id AND
+            awc_location_local.aggregation_level = agg_awc.aggregation_level AND
+            agg_awc.month = %(start_date)s
+        )
+        WHERE awc_location_local.aggregation_level=5 and awc_location_local.state_is_test<>1);
+        """.format(
+            tablename=self._tablename_func(),
+            columns=", ".join([col[0] for col in columns]),
+            calculations=", ".join([col[1] for col in columns])
+
+        ), {
+            'start_date': self.month_start
+        }
+
+    def update_queries(self):
+
+        yield """
+        CREATE UNLOGGED TABLE temp_gov_dashboard AS
```

Comment:
Why does this make a temp table then copy it directly to a real table? It seems like the second step does not add anything. Is that only because its necessary to get the data from the distributed to local table?

Reply:
yes. Else it gives error.